### PR TITLE
Added the asynchronous way of retrieving ui components from the registry

### LINKED ISSUE #7656: Add asynchronous variant for uiRegistry component in uiRegistry topic
# General issue

## Description:

This page is showing only synchronous variant of using uiRegistry component, while it's not good from performance perspective. In the JS file we have exampe how to retrieve any component using async variant: https://github.com/magento/magento2/blob/9544fb243d5848a497d4ea7b88e08609376ac39e/app/code/Magento/Ui/view/base/web/js/lib/registry/registry.js#L342-L346

## Possible solutions:

Add information about async way of retrieving ui components from registry and mark it as recommended.

### PR DESCRIPTION
**Pull request by chiranjeevi-cj**
## Purpose of this pull request

This pull request (PR) adds information about the async way of retrieving UI components from the registry and marked it as recommended.

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.3/ui_comp_guide/concepts/ui_comp_uiregistry.html

Fixes #7656 

whatsnew
Added an [example](https://devdocs.magento.com/guides/v2.4/ui_comp_guide/concepts/ui_comp_uiregistry.html) of asynchronously retrieving UI components from the registry.

### CODE REVIEW

File: src/guides/v2.3/ui_comp_guide/concepts/ui_comp_uiregistry.md [hunk 1]
```diff
@@ -37,6 +37,18 @@ To get a list of all components used on the current page, you can use this code:
 require('uiRegistry').get(function(component){console.log(component.name)});
 ```
 
+### Asynchronous way of retrieving UI components from the registry
+
+{:.bs-callout-info}
+This way is **Recommended**, for the good performance.
```
Comment (BarnyShergold) [inline src/guides/v2.3/ui_comp_guide/concepts/ui_comp_uiregistry.md:43]:
```suggestion
This is the **recommended** method for best performance.
```

File: src/guides/v2.3/ui_comp_guide/concepts/ui_comp_uiregistry.md [hunk 2]
```diff
@@ -37,6 +37,18 @@ To get a list of all components used on the current page, you can use this code:
 require('uiRegistry').get(function(component){console.log(component.name)});
 ```
 
+### Asynchronous way of retrieving UI components from the registry
```
Comment (BarnyShergold) [inline src/guides/v2.3/ui_comp_guide/concepts/ui_comp_uiregistry.md:40]:
```suggestion
### Asynchronous method of retrieving UI components from the registry
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Just a few wording suggestions

**Review by BarnyShergold — APPROVED** [review]
Excellent

**Comment by chiranjeevi-cj** [comment]
@BarnyShergold Sorry, I mistakenly requested you to review after you approved.

**Review by BarnyShergold — APPROVED** [review]
:)

**Review by jeff-matthews — APPROVED** [review]

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**